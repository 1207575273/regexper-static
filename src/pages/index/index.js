import React from 'react';
import ReactDOM from 'react-dom';

import App from '../../components/App';

import '../../style.css';
import '../../service-worker';

ReactDOM.render(<App/>, document.getElementById('root'));
