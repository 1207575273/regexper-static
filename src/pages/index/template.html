<!doctype html>
<html>
  <head>
    <meta charset="utf-8"/>
    <title>Regexper</title>
  </head>
  <body>
    <h1>App Content</h1>
  </body>
</html>
