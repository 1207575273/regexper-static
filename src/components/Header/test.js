import React from 'react';
import { shallow } from 'enzyme';

import { Header } from './index';

const env = { ...process.env };

afterEach(() => {
  process.env = env;
});

beforeEach(() => {
  process.env = {
    ...process.env,
    BANNER: 'testing'
  };
});

test('Header rendering', () => {
  const component = shallow(
    <Header>
      <p>Content</p>
    </Header>
  );
  expect(component).toMatchSnapshot();
});
