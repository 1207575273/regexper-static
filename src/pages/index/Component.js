import React from 'react';

import Message from 'components/Message';
import AlertIcon from 'feather-icons/dist/icons/alert-octagon.svg';
import Header from 'components/Header';
import Footer from 'components/Footer';
import App from 'components/App';

const Component = () => (
  <React.Fragment>
    <Header/>
    <noscript>
      <Message type="error" icon={ AlertIcon } heading="JavaScript Required">
        <p>You need to enable JavaScript to use Regexper.</p>
        <p>If you have concerns regarding the use of tracking code on Regexper, please see the <a href="/privacy.html">Privacy Policy</a>.</p>
      </Message>
    </noscript>
    <App/>
    <Footer/>
  </React.Fragment>
);

export default Component;
