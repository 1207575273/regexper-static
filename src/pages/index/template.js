import 'babel-register';
import React from 'react';

import '../../i18n';

import PageTemplate, { renderToString } from '../../components/PageTemplate';
import Message from '../../components/Message';
import AlertIcon from 'feather-icons/dist/icons/alert-octagon.svg';
import Header from '../../components/Header';
import Footer from '../../components/Footer';

export default renderToString(
  <PageTemplate>
    <Header/>
    <noscript>
      <Message className="error" icon={ AlertIcon } heading="JavaScript Required">
        <p>You need to enable JavaScript to use Regexper.</p>
        <p>Regexper and the tools used to create it are all open source. If you are concerned that the JavaScript being delivered is in any way malicious, please inspect the source by following the GitHub link in the header. There are two data collection tools integrated in the app:</p>
        <ul>
          <li><b>Google Analytics</b> is used to track browser usage data and application performance.</li>
          <li><b>Sentry.io</b> is a tool used to capture and report client-side JavaScript errors.</li>
        </ul>
        <p>Most popular ad blockers will prevent these tools from sending any tracking data, and doing so will <b>not</b> impact the performance of this app. Regexper is not supported by ad revenue or sales of any kind. The information collected by these tools is used to monitor application performance, determine browser support, and collect error reports.</p>
      </Message>
    </noscript>
    <Footer/>
  </PageTemplate>
);
