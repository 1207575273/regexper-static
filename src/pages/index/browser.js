import React from 'react';
import ReactDOM from 'react-dom';

import Component from './Component';
import RavenBoundary from '../../components/RavenBoundary';

import '../../style.css';
import '../../i18n';
import { setupServiceWorker } from '../../service-worker';
import { setupGA } from '../../analytics';
import { Raven, setupRaven } from '../../sentry';

setupRaven();

try {
  setupGA();
  if (process.env.NODE_ENV === 'production') {
    setupServiceWorker();
  }

  ReactDOM.render(
    <RavenBoundary>
      <Component/>
    </RavenBoundary>,
    document.getElementById('root'));
}
catch (e) {
  Raven.captureException(e);
}
