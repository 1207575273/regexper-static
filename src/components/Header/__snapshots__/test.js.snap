// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header rendering 1`] = `
<header
  className="header"
  data-banner="testing"
>
  <h1>
    <a
      href="/"
    >
      Regexper
    </a>
  </h1>
  <ul>
    <li>
      <a
        href="https://github.com/javallone/regexper-static"
        rel="external noopener noreferrer"
        target="_blank"
      >
        <SvgMock />
        <Trans>
          Source on GitHub
        </Trans>
      </a>
    </li>
  </ul>
</header>
`;
